sigstore-java/src/main/java/dev/sigstore/tuf/Updater.java

```diff
@@ -221,13 +224,22 @@ void verifyDelegate(
           } else {
             pubKey = Keys.constructTufPublicKey(Hex.decode(publicKeyContents), key.getScheme());
           }
-          byte[] signatureBytes = Hex.decode(signature.getSignature());
           try {
+            // while we error on keys that are not readable, we are intentionally more permissive
+            // about signatures. If for ANY reason (except unparsed keys) we cannot validate a
+            // signature, we continue as long as we find enough valid signatures within the
+            // threshold
+            byte[] signatureBytes = Hex.decode(signature.getSignature());
             if (verifiers.newVerifier(pubKey).verify(verificationMaterial, signatureBytes)) {
               goodSigs.add(signature.getKeyId());
             }
-          } catch (SignatureException e) {
-            throw new TufException(e);
+          } catch (SignatureException
+              | DecoderException
+              | NoSuchAlgorithmException
+              | InvalidKeyException e) {
+            log.fine(
+                "invalid signature '" + signature.getSignature() + "' for keyid '" + keyid + "'");
+            log.fine(e.getMessage());
```

Comment:
```suggestion
            log.log(Level.FINE,
                "invalid signature ''{0}}'' for keyid ''{1}''", new Object[]{signature.getSignature(), keyid, e});
```

Note the repeated single quotes `'` so they convert to a single quote in log.

Reply:
how does the throwable work here? Is there some magic that adds it to the message? Or do we need to use the logger than accepts throwable explicitly?

Reply:
maybe more like `log.log(Level.FINE, "invalid signature: " + sig + " for keyid: " + keyid ", e)`

Reply:
nvm, I reworked this a bit.